wsynphot/get_filter_data.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+import pandas as pd
+import requests
+
+from astropy.io.votable import parse_single_table
+# The VOTables fetched from SVO contain only single table element, thus parse_single_table
+
+SVO_MAIN_URL = 'http://svo2.cab.inta-csic.es/theory/fps/fps.php'
+
+# A utility fuction to convert byte strings to normal literal strings
+def byte_to_literal_strings(dataframe):
+    # Select the str columns:
+    str_df = dataframe.select_dtypes([np.object])
+    
+    # Convert all of them into unicode strings
+    str_df = str_df.stack().str.decode('utf-8').unstack()
+    
+    # Swap out converted cols with the original df cols
+    for col in str_df:
+        dataframe[col] = str_df[col]
+    return dataframe
+
+
+def get_filter_index():
+    # To get master list, fetch all filters from SVO of wavelength between 0 & very large no.
+    query = {'WavelengthEff_min': 0,
+             'WavelengthEff_max': 10000000}
+    response = requests.get(SVO_MAIN_URL, params=query)
+    response.raise_for_status()
+    
+    # Parse the fetched VOTable (XML file) into an Astropy table
+    votable = io.BytesIO(response.content)
+    table = parse_single_table(votable).to_table()
```

Comment:
let's return the votable.

Reply:
Okay!
What I was thinking is to remove this parsing from each of these 3 functions, as you said. And put it in another function `convert_to_dataframe(votable)` - this will let them see the work we have done in advance to parse VOTables into dataframes (including conversion of byte strings in dataframe to normal ones). What do you think?

Reply:
Done